### PR TITLE
chore: bump go version

### PR DESCRIPTION
**Pull request by bharathkkb**

### CODE REVIEW

File: go.work [hunk 1]
```diff
@@ -1,4 +1,4 @@
-go 1.21
+go 1.22
```
Comment (apeabody) [inline go.work:1]: The dev-tools image currently includes GO 1.21

Comment (bharathkkb) [reply go.work:1]: Ah thanks for catching! I guess we are using alpine go which still seems to be 1.21. This is not urgent and we can hold off-  I was just experimenting with something that needed 1.22 so I split this off into a PR.

Comment (apeabody) [reply go.work:1]: Yeah - The current Alpine v3.19 has GO v1.21.  Looks like Alpine v3.20 is targeted for ~June.

Comment (apeabody) [reply go.work:1]: Go 1.22 included in dev-tools 1.21

**Review by apeabody — APPROVED** [review]
LGTM

**Pull request merged**